crypto/evp/evp_fetch.c

```diff
@@ -343,10 +344,9 @@ inner_evp_generic_fetch(OPENSSL_CTX *libctx, int operation_id,
             if (name_id == 0)
                 name_id = ossl_namemap_name2num(namemap, name);
             meth_id = evp_method_id(operation_id, name_id);
-            ossl_method_store_cache_set(store, meth_id, properties, method);
+            ossl_method_store_cache_set(store, meth_id, properties, method,
+                                        up_ref_method, free_method);
         }
-    } else {
-        up_ref_method(method);
```

Comment:
I was surprised by this, but seeing the rest of the change, I get it, the recount responsibility has moved entirely to the store & cache code.

Reply:
This leads to double free problems.
It does solve the current leaked memory problems.

There is something fundamental I'm missing & my brain isn't working well at present.

Reply:
oops, wrong place for this comment :(

Reply:
Figured it out, missed a ref counter update.

But yes, the store and cache are responsible for the reference counting now.  A return from either will have the count increased.